node/src/store/indexed_db_store.rs

```diff
@@ -214,6 +229,11 @@ impl IndexedDbStore {
             header_store.add(&jsvalue_header, None).await?;
         }
 
+        header_ranges.insert_relaxed(headers_range)?;
+        set_ranges(&ranges_store, HEADER_RANGES_KEY, &header_ranges).await?;
+
+        tx.commit().await?;
+
         if tail.height().value()
```

Comment:
any particular reason for changing the order here?

Reply:
Yes. If `commit()` fails with an error, then `self.head` should stay as it was before. By changing the order we make this possible.